src/rv/comm/rcssserver/GameState.java

```diff
@@ -48,6 +48,34 @@
         void gsServerMessageProcessed(GameState gs);
     }
 
+    public enum FoulType {
+        CROWDING(0, "crowding"), TOUCHING(1, "touching"), ILLEGALDEFENCE(2, "illegal defence"),
+        ILLEGALATTACK(3, "illegal attack"), INCAPABLE(4, "incapable"),
+        KICKOFF(5, "illegal kickoff"), CHARGING(6, "charging");
+
+        private int    idx;
```

Comment:
Hmm I guess so.  At this point `idx` is perhaps just notation in the code letting you know a `FoulType`'s index in  the enum array which will be returned when `values()` is called.